processor/logcountprocessor/config.go

```diff
@@ -0,0 +1,85 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package logcountprocessor provides a processor that counts logs as metrics.
+package logcountprocessor
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/antonmedv/expr"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+)
+
+const (
+	// defaultMetricName is the default metric name.
+	defaultMetricName = "log.count"
+
+	// defaultMetricUnit is the default metric unit.
+	defaultMetricUnit = "{logs}"
+
+	// defaultInterval is the default metric interval.
+	defaultInterval = time.Minute
+
+	// defaultMatch is the default match expression.
+	defaultMatch = "true"
+)
+
+// Config is the config of the processor.
+type Config struct {
+	config.ProcessorSettings `mapstructure:",squash"`
+	Route                    string            `mapstructure:"route"`
```

Comment:
Can/do we need to do any validation that this actually exists?

Reply:
Unfortunately not. This config will have no knowledge of outside components or configured pipelines. Receivers are also instantiated last in the pipeline. So at this point, there is no guarantee the receiver has been instantiated.

Reply:
Ok, looks like we'll probably log an error from the processor if it tries to route somewhere that doesn't exist?

Reply:
Yeah exactly. We can't validate ahead of time. But we can log and warn the user when they attempt to route to something that doesn't exist.